### PR TITLE
Required Column added

### PR DESCRIPTION
**Pull request by souravsk**
Create the PR for this issue:-
Add information about optional/required fields in Actions documentation #380

### CODE REVIEW

File: docs/actions.md [hunk 1]
```diff
@@ -49,19 +49,19 @@ Follow these steps to set up JIRA integration:
...
+|*password*    | Optional: User's password. API token can also be used for Cloud Jira instances. |             | NO          |    
+|*token*       | Optional: User's Personal Access Token. Used only for Jira Server/Data Center   |             | NO          |
+|*board*       |  Optional: JIRA board key              |           | NO          |
+|*priority*    |  Optional: ticket priority, e.g., High |           | NO          |
+|*assignee*    | Optional: comma separated list of users (emails) that will be assigned to ticket, e.g., ["john@yahoo.com"]. To assign a ticket to the Application Owner email address (as defined in Aqua Application Scope, owner email field), specify ["<%application_scope_owner%>"] as the assignee value   |               | NO          | 
+|*issuetype*   | Optional: issue type, e.g., Bug        |           | NO          | 
+|*labels*      | Optional: comma separated list of labels that will be assigned to ticket, e.g., ["label1", "label2"]|         | NO          |  
+|*sprint*      | Optional: Sprint name, e.g., "3.5 Sprint 8" |      | NO          |
```
Comment (simar7) [inline docs/actions.md:64]: As mentioned, can we remove the "optional" and "required" strings from the description column?

File: docs/actions.md [hunk 2]
```diff
@@ -169,29 +169,29 @@ You will need to care about an HTTP Event Collector in Splunk Enterprise or Splu
...
-*user* | ServiceNow user name |
-*password* | User API key / password |
-*instance* | Name of ServiceNow Instance (usually the XXX at XXX.servicenow.com)|
-*board* | ServiceNow board name to open tickets on. Default is "incident" |
+|Key        | Description                                                        | Possible Values  | Required  |
+|---------- | ------------------------------------------------------------------ | ---------------  | --------- | 
+|*user*     | ServiceNow user name                                               |                  | YES       |            
+|*password* | User API key / password                                            |                  | YES       |
+|*instance* | Name of ServiceNow Instance (usually the XXX at XXX.servicenow.com)|                  | YES       |
+|*board*    | ServiceNow board name to open tickets on. Default is "incident"    |                  |           |
```
Comment (simar7) [inline docs/actions.md:185]: Board is not required.

File: docs/actions.md [hunk 3]
```diff
@@ -214,24 +214,24 @@ Key | Description | Possible Values
...
-| Option      | Usage                                                                                     |
-|-------------|-------------------------------------------------------------------------------------------|
-| env         | Optional, custom environment variables to be exposed in the shell of the executing script |
-| input-file  | Required, custom shell script to executed                                                 |
-| exec-script | Required, inline shell script executed                                                    |
+| Option      | Usage                                                                                     | Required |
+|-------------|-------------------------------------------------------------------------------------------|----------|
+| env         | Optional, custom environment variables to be exposed in the shell of the executing script | NO       |
+| input-file  | Required, custom shell script to executed                                                 | YES      |
+| exec-script | Required, inline shell script executed                                                    | YES      |
```
Comment (simar7) [inline docs/actions.md:234]: ditto on removing the optional/required text from usage description.

File: docs/actions.md [hunk 4]
```diff
@@ -243,36 +243,36 @@ Below is an example of using `$POSTEE_EVENT`. It uses the inline exec-script scr
...
-| Headers  | Optional, custom headers to send        |
-| Timeout  | Optional, custom timeout for HTTP call  |
-| Bodyfile | Optional, input file for HTTP post body |
+| Option   | Usage                                   | Required |
+|----------|-----------------------------------------|----------|
+| URL      | Required, URL of the remote server      | YES      |
+| Method   | Required, e.g., GET, POST               | YES      |
+| Headers  | Optional, custom headers to send        | NO       |
+| Timeout  | Optional, custom timeout for HTTP call  | NO       |
+| Bodyfile | Optional, input file for HTTP post body | NO       |
```
Comment (simar7) [inline docs/actions.md:252]: ditto on removing the optional/required text from usage description.

File: docs/actions.md [hunk 5]
```diff
@@ -243,36 +243,36 @@ Below is an example of using `$POSTEE_EVENT`. It uses the inline exec-script scr
...
-| kube-actions        | Optional, key-value pair of labels and annotations<br/>Labels must be added via "labels" key and Annotations via "annotations". |
+| Option              | Usage                                                                                                                           | Required |
+|---------------------|---------------------------------------------------------------------------------------------------------------------------------|----------|
+| kube-namespace      | Required. Kubernetes namespace to use.                                                                                          | YES      |
+| kube-config-file    | Required. Path to .kubeconfig file                                                                                              | YES      |
+| kube-label-selector | Required, if specifying labels or annotations.                                                                                  | YES      |
+| kube-actions        | Optional, key-value pair of labels and annotations<br/>Labels must be added via "labels" key and Annotations via "annotations". | NO       |
```
Comment (simar7) [inline docs/actions.md:263]: ditto on removing the optional/required text from usage description.

File: docs/actions.md [hunk 6]
```diff
@@ -243,36 +243,36 @@ Below is an example of using `$POSTEE_EVENT`. It uses the inline exec-script scr
...
+| docker-image-name    | Required. Image name of the docker image.                                                                                                                                | YES      |
+| docker-cmd           | Required. Command to run inside the docker image.                                                                                                                        | YES      |
+| docker-env           | Optional. Environment variables to set in the container.                                                                                                                 | NO       |
+| docker-network       | Optional. Connect the action container to the specified network. {e.g. "host"}                                                                                           | NO       |
+| docker-volume-mounts | Optional*. Volume mounts present inside the container.<br/> * _If you have specified volume mounts, you also need to pass them through into the postee docker container_ | NO       |
```
Comment (simar7) [inline docs/actions.md:275]: ditto on removing the optional/required text from usage description.

**Review by simar7 — CHANGES_REQUESTED** [review]
thanks it looks awesome, just suggested a few changes.

File: docs/actions.md [hunk 7]
```diff
@@ -49,19 +49,19 @@ Follow these steps to set up JIRA integration:
...
-*user* | Jira user. Use email for Jira Cloud and UserName for Jira Server/Data Center |
-*password* | Optional: User's password. API token can also be used for Cloud Jira instances. |
-*token* | Optional: User's Personal Access Token. Used only for Jira Server/Data Center |
-*board* |  Optional: JIRA board key |
-*priority*|  Optional: ticket priority, e.g., High |
-*assignee*| Optional: comma separated list of users (emails) that will be assigned to ticket, e.g., ["john@yahoo.com"]. To assign a ticket to the Application Owner email address (as defined in Aqua Application Scope, owner email field), specify ["<%application_scope_owner%>"] as the assignee value |
-*issuetype*| Optional: issue type, e.g., Bug |
-*labels*| Optional: comma separated list of labels that will be assigned to ticket, e.g., ["label1", "label2"]|
-*sprint*| Optional: Sprint name, e.g., "3.5 Sprint 8" |
+Key           | Description          | Possible Values | Required
```
Comment (krol3) [inline docs/actions.md:52]:
```suggestion
Key           | Description          | Supported Values | Required 
```
Or

```suggestion
Key           | Description          | Values | Required 
```
@simar7  I think the word `Possible` is a little ambiguos. What do you think?

File: docs/actions.md [hunk 8]
```diff
@@ -49,19 +49,19 @@ Follow these steps to set up JIRA integration:
...
-*token* | Optional: User's Personal Access Token. Used only for Jira Server/Data Center |
-*board* |  Optional: JIRA board key |
-*priority*|  Optional: ticket priority, e.g., High |
-*assignee*| Optional: comma separated list of users (emails) that will be assigned to ticket, e.g., ["john@yahoo.com"]. To assign a ticket to the Application Owner email address (as defined in Aqua Application Scope, owner email field), specify ["<%application_scope_owner%>"] as the assignee value |
-*issuetype*| Optional: issue type, e.g., Bug |
-*labels*| Optional: comma separated list of labels that will be assigned to ticket, e.g., ["label1", "label2"]|
-*sprint*| Optional: Sprint name, e.g., "3.5 Sprint 8" |
+Key           | Description          | Possible Values | Required 
+--------------|----------------------|-----------------|----------
+*url*         | Jira project url     |                 | YES
```
Comment (krol3) [inline docs/actions.md:54]:
```suggestion
*url*         | Jira project url     |                 | Yes      
```
I think it's better use lowercase.

**Review by krol3 — COMMENTED** [review]
Thank you for your contribution @souravsk , a few comments below

File: docs/actions.md (hunk 8 above)
Comment (souravsk) [reply docs/actions.md:54]: okay @krol3 
1:- "Yes"
2:- "yes"
which one will be good

File: docs/actions.md (hunk 7 above)
Comment (simar7) [reply docs/actions.md:52]: yeah your suggestion makes sense

Comment (souravsk) [reply docs/actions.md:52]: So which should I select

Comment (simar7) [reply docs/actions.md:52]: I'd leave it at Values.

File: docs/actions.md (hunk 8 above)
Comment (simar7) [reply docs/actions.md:54]: I'd pick 1.

File: docs/actions.md (hunk 7 above)
Comment (souravsk) [reply docs/actions.md:52]: okay

**Review by simar7 — APPROVED** [review]
Thanks for your contribution.

**Comment by souravsk** [comment]
Welcome @simar7 if any else I do please let me know

**Comment by simar7** [comment]
> Welcome @simar7 if any else I do please let me know

thanks @souravsk - I'm in the process of writing a design document that should highlight where we can improve Postee today. It's currently a PR in progress. I plan to create actionable issues from this document, that external contributors like you can pick up and work on. https://github.com/aquasecurity/postee/pull/466/files

**Comment by souravsk** [comment]
it looks like it is already merged

**Pull request merged**